src/ec-evaluator/interpreter.ts

```diff
@@ -249,9 +261,11 @@ function runECEMachine(context: Context, agenda: Agenda, stash: Stash, isPrelude
     if (agenda.isEmpty() && stash.isEmpty()) {
       stash.push(undefined)
     }
-
     command = agenda.peek()
-    steps += 1
+    // If command is an ImportDeclaration, don't add unnecessary step
+    if (!isImportDeclaration) {
```

Comment:
I agree that this seems a bit hacky, and also feels a bit like its going against the concept of the agenda. Another problem with this is that at step 2 (in the screenshot) the user thinks that the import statement is being evaluated, but in step 3, another command is evaluated instead.

Reply:
Suggestion: In 'Program' when the block is decomposed into individual statements and pushed onto the agenda, don't push any import statements since the imports are already evaluated in 'Program'. (I would think modifying the 'handleSequence' function appropriately would work best). Then we wouldn't need this hack since our evaluator won't encounter any import statements.

Reply:
Rest looks good to me!

Reply:
> Suggestion: In 'Program' when the block is decomposed into individual statements and pushed onto the agenda, don't push any import statements since the imports are already evaluated in 'Program'. (I would think modifying the 'handleSequence' function appropriately would work best). Then we wouldn't need this hack since our evaluator won't encounter any import statements.

this means that the import statements won't be in the agenda at all?

Reply:
Yep! The user can see the import statements initially in the program block, but since the import statements are evaluated in the program block itself it doesn't seem to make sense to continue to push the import statements on the agenda since they do nothing and only add to confusion. 
Do you agree prof? @martin-henz

Reply:
Yes, I agree. The handling of the program needs to create the program environment which will include the imported names and the newly declared names. The imported names can be immediately assigned to their implementation without the need of executing a step in the machine.

Reply:
okay, i have made the necessary changes to the handleSequence function so the import statements are only in the program block but not on the agenda

<img width="949" alt="Screenshot 2023-07-27 at 2 39 03 PM" src="https://github.com/source-academy/js-slang/assets/77202894/2a31ebd0-90fa-4429-bf41-b14ab33faad1">

<img width="930" alt="Screenshot 2023-07-27 at 2 39 43 PM" src="https://github.com/source-academy/js-slang/assets/77202894/bf67b492-29c9-46e5-93f2-ac8fa062468b">